### PR TITLE
Enable STM32WB I2C

### PR DESCRIPTION
**Pull request by rogernguessan**
This contribution enables `STM32WB` `I2C`

### CODE REVIEW

File: soc/arm/st_stm32/stm32wb/dts_fixup.h [hunk 1]
```diff
@@ -83,5 +83,24 @@
 #define DT_FLASH_DEV_BASE_ADDRESS		DT_ST_STM32WB_FLASH_CONTROLLER_58004000_BASE_ADDRESS
 #define DT_FLASH_DEV_NAME			DT_ST_STM32WB_FLASH_CONTROLLER_58004000_LABEL
 
+#define DT_I2C_1_BASE_ADDRESS               DT_ST_STM32_I2C_V2_40005400_BASE_ADDRESS
```
Comment (ydamigos) [inline soc/arm/st_stm32/stm32wb/dts_fixup.h]: Fix alignment.

**Review by ydamigos — CHANGES_REQUESTED** [review]
Looks good. I don't have the hardware to test it. Some minor comments.

File: dts/arm/st/wb/stm32wb.dtsi [hunk 2]
```diff
@@ -122,6 +123,32 @@
 			label = "UART_1";
 		};
 
+    i2c1: i2c@40005400 {
```
Comment (erwango) [inline dts/arm/st/wb/stm32wb.dtsi]: Indent this line to same (align on other nodes)

File: dts/arm/st/wb/stm32wb.dtsi [hunk 3]
```diff
@@ -122,6 +123,32 @@
...
+			#size-cells = <0>;
+			reg = <0x40005400 0x400>;
+			clocks = <&rcc STM32_CLOCK_BUS_APB1 0x00200000>;
+			interrupts = <30 0>, <31 0>;
+			interrupt-names = "event", "error";
+			status = "disabled";
+			label= "I2C_1";
+		};
+
+    i2c3: i2c@40005c00 {
```
Comment (erwango) [inline dts/arm/st/wb/stm32wb.dtsi]: ditto

**Review by erwango — CHANGES_REQUESTED** [review]
Looks good, some minor comments

**Review by erwango — CHANGES_REQUESTED** [review]
Miss board doc update as well

**Review by dbkinder — APPROVED** [review]
doc changes LGTM.

File: boards/arm/nucleo_wb55rg/doc/nucleowb55rg.rst [hunk 4]
```diff
@@ -178,6 +178,10 @@ Default Zephyr Peripheral Mapping:
 
 - UART_1 TX/RX : PB7/PB6
 - LPUART_1 TX/RX : PA3/PA2 (arduino_serial)
+- I2C_1_SCL : PB8
```
Comment (ydamigos) [inline boards/arm/nucleo_wb55rg/doc/nucleowb55rg.rst]: Please add I2C in Supported Features table.

Comment (rgf97) [reply boards/arm/nucleo_wb55rg/doc/nucleowb55rg.rst]: Thanks for your precious eyes :)

**Review by erwango — APPROVED** [review]

**Review by ydamigos — APPROVED** [review]

**Pull request merged**